### PR TITLE
Scrum 53 API DEVELOPMENT asset position history

### PR DESCRIPTION
**Pull request by dskudar20**
Implemented AssetPositionHistory class, AssetPositionHistory repository, AssetPositionHistory service and AssetPositionHistory controller.

### CODE REVIEW

**Review by rikjalzabet — APPROVED** [review]

File: BusinessLogicLayer/Services/AssetPositionHistoryService.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            _assetPositionHistoryRepository = assetPositionHistoryRepository;
+        }
+
+        public async Task<bool> AddAssetPositionHistory(AssetPositionHistory assetPositionHistory)
+        {
+            bool isSuccessful = false;
+            int affectedRow = await _assetPositionHistoryRepository.AddAssetPositionHistory(assetPositionHistory);
+
+            isSuccessful = affectedRow > 0;
+            return isSuccessful;
```
Comment (ahamzic20) [inline BusinessLogicLayer/Services/AssetPositionHistoryService.cs]: isSuccesfull not needed, use `return affected row > 0 `

File: BusinessLogicLayer/Services/AssetService.cs [hunk 2]
```diff
@@ -13,17 +14,17 @@ public class AssetService : IAssetService
...
             return queryableAssets.ToList();
         }
         public async Task<bool> AddAsset(Asset asset)
         {
             bool isSuccessful = false;
-            int affectedRow = await _assetRepository.AddAsset(asset);
+            int affectedRow = await _assetRepository.AddAsync(asset);
 
             isSuccessful = affectedRow > 0;
             return isSuccessful;
```
Comment (ahamzic20) [inline BusinessLogicLayer/Services/AssetService.cs]: also here

File: BusinessLogicLayer/Services/FloorMapService.cs [hunk 3]
```diff
@@ -18,7 +19,7 @@ public FloorMapService(IFloorMapRepository floorMapRepository)
         public async Task<bool> AddFloorMap(FloorMap floorMap)
         {
             bool isSuccessful = false;
-            int affectedRow = await _floorMapRepository.AddFloorMap(floorMap);
+            int affectedRow = await _floorMapRepository.AddAsync(floorMap);
 
             isSuccessful = affectedRow > 0;
             return isSuccessful;
```
Comment (ahamzic20) [inline BusinessLogicLayer/Services/FloorMapService.cs]: Also here

File: BusinessLogicLayer/Services/ZoneService.cs [hunk 4]
```diff
@@ -17,20 +18,20 @@ public ZoneService(IZoneRepository zoneRepository)
...
             return zone;
         }
 
         public async Task<bool> AddZone(Zone zone)
         {
             bool isSuccessful = false;
-            int affectedRow = await _zoneRepository.AddZone(zone);
+            int affectedRow = await _zoneRepository.AddAsync(zone);
 
             isSuccessful = affectedRow > 0;
```
Comment (ahamzic20) [inline BusinessLogicLayer/Services/ZoneService.cs]: also here

File: DataAccessLayer/AppDbContext.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        public AppDbContext(DbContextOptions<AppDbContext> options)
+      : base(options)
+        {
+        }
+        public DbSet<Zone> Zones { get; set; }
+        public DbSet<Asset> Assets { get; set; }
+        public DbSet<FloorMap> FloorMaps { get; set; }
+
+        public DbSet<AssetPositionHistory> AssetPositionHistory { get; set; }
+        //public DbSet<AssetZoneHistory> AssetZoneHistory { get; set; }
```
Comment (ahamzic20) [inline DataAccessLayer/AppDbContext.cs:24]: is this comment needed?

File: DataAccessLayer/AppDbContext.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                entity.HasKey(a => a.Id);
+                entity.HasOne(a => a.Asset)
+                      .WithMany()
+                      .HasForeignKey(a => a.AssetId)
+                      .OnDelete(DeleteBehavior.Cascade);
+
+                entity.HasOne(a => a.Zone)
+                      .WithMany()
+                      .HasForeignKey(a => a.ZoneId);
+            });
```
Comment (ahamzic20) [inline DataAccessLayer/AppDbContext.cs:79]: Also is this comment needed

File: EntityLayer/Entities/Asset.cs [hunk 7]
```diff
@@ -15,5 +16,10 @@ public class Asset:IAsset
         public DateTime LastSync { get; set; }
         public int FloorMapId { get; set; }
         public bool Active { get; set; }
+
+        public virtual FloorMap FloorMap { get; set; }
+        //public virtual ICollection<AssetPositionHistory> AssetPositionHistories { get; set; } = new List<AssetPositionHistory>();
```
Comment (ahamzic20) [inline EntityLayer/Entities/Asset.cs:21]: comment?

File: EntityLayer/Entities/FloorMap.cs [hunk 8]
```diff
@@ -11,5 +11,9 @@ public class FloorMap
         public int Id { get; set; }
         public string Name { get; set; }
         public string Image { get; set; }
+
+        //public virtual ICollection<Asset> Assets { get; set; } = new List<Asset>();
```
Comment (ahamzic20) [inline EntityLayer/Entities/FloorMap.cs:15]: comment?

File: EntityLayer/Entities/Point.cs [hunk 9]
```diff
@@ -10,5 +10,8 @@ public class Point
     {
         public double X { get; set; }
         public double Y { get; set; }
+        //public int ZoneId { get; set; }
```
Comment (ahamzic20) [inline EntityLayer/Entities/Point.cs:13]: comment?

**Review by ahamzic20 — CHANGES_REQUESTED** [review]
See comments and change if needed.

File: DataAccessLayer/AppDbContext.cs (hunk 5 above)
Comment (dskudar20) [reply DataAccessLayer/AppDbContext.cs:24]: This code will be used after creating the AssetZoneHistory class.

File: DataAccessLayer/AppDbContext.cs (hunk 6 above)
Comment (dskudar20) [reply DataAccessLayer/AppDbContext.cs:79]: This code will be used after creating the AssetZoneHistory class.

File: EntityLayer/Entities/Asset.cs (hunk 7 above)
Comment (dskudar20) [reply EntityLayer/Entities/Asset.cs:21]: This code will be used after creating the AssetZoneHistory class.

File: EntityLayer/Entities/FloorMap.cs (hunk 8 above)
Comment (dskudar20) [reply EntityLayer/Entities/FloorMap.cs:15]: This code will be used after creating the AssetZoneHistory class.

File: EntityLayer/Entities/Point.cs (hunk 9 above)
Comment (dskudar20) [reply EntityLayer/Entities/Point.cs:13]: This code will be used after creating the AssetZoneHistory class.

**Review by ahamzic20 — APPROVED** [review]
Fixed with changes.

**Pull request merged**